pdg/src/snapshots/c2rust_pdg__tests__analysis_test_pdg_snapshot_debug.snap

```diff
@@ -821,112 +835,134 @@ nodes_that_need_write = [3, 2, 1, 0]
 
 g {
 	n[0]: alloc       _    => _1   @ bb1[2]:  fn test_load_value_store_value;  _1 = malloc(move _2);
-	n[1]: value.load  _    => _5   @ bb2[6]:  fn test_load_value_store_value;  _5 = (*_4);
-	n[2]: value.store n[1] => _4.* @ bb2[7]:  fn test_load_value_store_value;  (*_4) = move _5;
-	n[3]: value.load  _    => _7   @ bb2[11]: fn test_load_value_store_value;  _7 = (*_4);
-	n[4]: free        n[3] => _6   @ bb2[12]: fn test_load_value_store_value;  _6 = free(move _7);
+	n[1]: value.load  _    => _5   @ bb2[7]:  fn test_load_value_store_value;  _5 = (*_4);
+	n[2]: value.store n[1] => _4.* @ bb2[8]:  fn test_load_value_store_value;  (*_4) = move _5;
+	n[3]: value.load  _    => _7   @ bb2[12]: fn test_load_value_store_value;  _7 = (*_4);
+	n[4]: free        n[3] => _6   @ bb2[13]: fn test_load_value_store_value;  _6 = free(move _7);
+}
+nodes_that_need_write = []
+
+g {
+	n[0]: &_1        _    => _  @ bb2[0]:  fn test_load_value_store_value;  _8 = &raw mut _1;
+	n[1]: copy       n[0] => _4 @ bb2[4]:  fn test_load_value_store_value;  _4 = &raw mut (*_8);
+	n[2]: addr.load  n[1] => _  @ bb2[7]:  fn test_load_value_store_value;  _5 = (*_4);
+	n[3]: addr.store n[1] => _  @ bb2[8]:  fn test_load_value_store_value;  (*_4) = move _5;
+	n[4]: addr.load  n[1] => _  @ bb2[12]: fn test_load_value_store_value;  _7 = (*_4);
+}
+nodes_that_need_write = [3, 1, 0]
+
+g {
+	n[0]:  &_35       _     => _   @ bb30[4]:  fn main_0;          _49 = &raw mut _35;
+	n[1]:  copy       n[0]  => _34 @ bb30[5]:  fn main_0;          _34 = &mut (*_49);
+	n[2]:  addr.store n[0]  => _   @ bb30[3]:  fn main_0;          _35 = [const 2_i32, const 5_i32, const 3_i32, const 1_i32, const 6_i32];
+	n[3]:  copy       n[1]  => _40 @ bb30[12]: fn main_0;          _40 = &(*_34);
```

Comment:
The ordering here seems off - I think nodes 1 and 2 are swapped.  Usually for initialization of an address-taken local, you'll see something like `&_1 @ bb0[2]; addr.store @ bb0[1]`, where the store that initializes the local comes first in the MIR but second in the PDG.  Here, there's a third node in between (`n[1]`).  Also, this ordering introduces a spurious conflict: the pointer produced by `n[1]` is live across the store in `n[2]` (note there are things below with `source: n[1]`).

Reply:
You're right, fixed in https://github.com/immunant/c2rust/pull/654/commits/b86ccc7e9154870bf0c0abf0dc681407e10d96f8 which includes the fix to the snapshot.